wire-ios-request-strategy/Sources/E2EIdentity/CRL/CertificateRevocationListAPI.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireDataModel
+
+public class CertificateRevocationListAPI: CertificateRevocationListAPIProtocol {
+    private let httpClient: HttpClientCustom
+
+    public init(httpClient: HttpClientCustom = HttpClientE2EI()) {
+        self.httpClient = httpClient
+    }
+
+    public func getRevocationList(from distributionPoint: URL) async throws -> Data {
+        var request = URLRequest(url: distributionPoint)
+        request.httpMethod = "GET"
+        let (data, _) = try await httpClient.send(request)
+
+        return data
```

Comment:
Should we check the HTTP status code if it is 2xx or an error, before returning data?

Reply:
done https://github.com/wireapp/wire-ios/pull/922/commits/93d4a3ba008dc955e51ee7f3be16ba7f241f5435